frontend/lib/resolution/default-functions.cpp

```diff
@@ -310,6 +288,53 @@ static bool initHelper(Context* context,
     stmts.push_back(std::move(assign));
 
     formals.push_back(std::move(formal));
+  };
+
+  std::function<void (const AstNode*)> processDecl = [&addFormal, &processDecl](auto ast) {
+    if (auto var = ast->toVarLikeDecl()) {
+      addFormal(var, var->typeExpression(), var->initExpression());
+    } else if (auto multi = ast->toMultiDecl()) {
+      auto it = multi->decls().begin();
+      auto groupBegin = it;
+      const AstNode* curTypeExpr = nullptr;
+      const AstNode* curInitExpr = nullptr;
+      while (it != multi->decls().end()) {
```

Comment:
It'd be nice to comment here that this loop is iterating over the 1st variable in a multidecl that has its type or init expr set by a later variable.

Reply:
It's actually the opposite: We iterate until we find a decl with either a type or init expression, then the inner loop goes back and re-uses the later variable's information as necessary.

So in ``var a, b, c: int``, it does nothing the first two iterations, and then in the third iteration will use ``int`` from ``c``.